core/src/main/java/com/sequenceiq/cloudbreak/rotation/executor/CMUserRotationExecutor.java

```diff
@@ -0,0 +1,95 @@
+import com.sequenceiq.cloudbreak.dto.StackDto;
+import com.sequenceiq.cloudbreak.rotation.context.CMUserRotationContext;
+import com.sequenceiq.cloudbreak.rotation.secret.RotationContext;
+import com.sequenceiq.cloudbreak.rotation.secret.RotationExecutor;
+import com.sequenceiq.cloudbreak.rotation.secret.SecretLocationType;
+import com.sequenceiq.cloudbreak.rotation.secret.SecretRotationException;
+import com.sequenceiq.cloudbreak.service.CloudbreakException;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterApiConnectors;
+import com.sequenceiq.cloudbreak.service.secret.domain.RotationSecret;
+import com.sequenceiq.cloudbreak.service.secret.service.SecretService;
+import com.sequenceiq.cloudbreak.service.stack.StackDtoService;
+
+@Component
+public class CMUserRotationExecutor implements RotationExecutor<CMUserRotationContext> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CMUserRotationExecutor.class);
+
+    @Inject
+    private ClusterApiConnectors clusterApiConnectors;
+
+    @Inject
+    private StackDtoService stackService;
+
+    @Inject
+    private SecretService secretService;
+
+    @Override
+    public void rotate(CMUserRotationContext rotationContext) {
+        LOGGER.info("Starting rotation of CM user for resource {} by creating a new user.", rotationContext.getResourceCrn());
+        ClusterSecurityService clusterSecurityService = getClusterSecurityService(rotationContext);
+        RotationSecret userRotationSecret = secretService.getRotation(rotationContext.getUserSecret());
+        RotationSecret passwordRotationSecret = secretService.getRotation(rotationContext.getPasswordSecret());
+        String clientUser = secretService.get(rotationContext.getClientUserSecret());
+        String clientPassword = secretService.get(rotationContext.getClientPasswordSecret());
+        try {
+            clusterSecurityService.createNewUser(
+                    userRotationSecret.getBackupSecret(),
+                    userRotationSecret.getSecret(),
+                    passwordRotationSecret.getSecret(),
+                    clientUser,
+                    clientPassword);
+        } catch (CloudbreakException e) {
+            LOGGER.error("Rotation of CM user failed, ", e);
+            throw new SecretRotationException(e, getType());
+        }
+    }
+
+    @Override
+    public void rollback(CMUserRotationContext rotationContext) {
+        LOGGER.info("Starting to rollback rotation of CM user for resource {} by deleting the new user", rotationContext.getResourceCrn());
+        RotationSecret userRotationSecret = secretService.getRotation(rotationContext.getUserSecret());
+        deleteUser(userRotationSecret.getSecret(), rotationContext);
+    }
+
+    @Override
+    public void finalize(CMUserRotationContext rotationContext) {
+        LOGGER.info("Finalizing rotation of CM user for resource {} by deleting the old user", rotationContext.getResourceCrn());
```

Comment:
I would check if we can reach the cluster with the new user and only then delete old one. Also the isRotation() check can be usefull to only delete if there is an old user.